app/services/objective_notification_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class ObjectiveNotificationService
+  def perform
+    Objective.not_archived.each do |objective|
+      check_almost_achieved(objective, WeekToGoalAction.new.calculate(objective))
+      check_achieved(objective, objective.account.balance, objective.amount)
+    end
+  end
+
+  private
+
+  def check_almost_achieved(objective, total_weeks_left)
```

Comment:
The main responsibility of this method is not checking. It actually **sends** almost achieved emails. The method name should be changed respectively.